crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -102,27 +115,77 @@ impl SettlementEncoder {
         let sell_token_index = self
             .token_index(order.order_creation.sell_token)
             .expect("missing sell token with price");
+        let (trade, execution) = match is_liquidity_order {
```

Comment:
Since most of the function is different based on whether you add a liquidity order, could it be nicer to have 2 separate functions instead?